models/spring-ai-bedrock-converse/src/test/java/org/springframework/ai/bedrock/converse/client/BedrockNovaChatClientIT.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ClassPathResource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Christian Tzolov
+ */
+@SpringBootTest(classes = BedrockNovaChatClientIT.Config.class)
+@EnabledIfEnvironmentVariable(named = "AWS_ACCESS_KEY_ID", matches = ".*")
+@EnabledIfEnvironmentVariable(named = "AWS_SECRET_ACCESS_KEY", matches = ".*")
+@EnabledIfEnvironmentVariable(named = "AWS_SESSION_TOKEN", matches = ".*")
+public class BedrockNovaChatClientIT {
+
+	private static final Logger logger = LoggerFactory.getLogger(BedrockNovaChatClientIT.class);
+
+	@Autowired
+	ChatModel chatModel;
+
+	@Test
+	void pdfMultiModalityTest() throws IOException {
+
+		String response = ChatClient.create(chatModel)
+			.prompt()
+			.user(u -> u.text(
+					"You are a very professional document summarization specialist. Please summarize the given document.")
+				.media(Media.Format.DOC_PDF, new ClassPathResource("/spring-ai-reference-overview.pdf")))
+			.call()
+			.content();
+
+		logger.info(response);
+		assertThat(response).containsAnyOf("Spring AI", "portable API");
+	}
+
+	@Test
+	void imageMultiModalityTest() throws IOException {
+
+		String response = ChatClient.create(chatModel)
+			.prompt()
+			.user(u -> u.text("Explain what do you see on this picture?")
+				.media(Media.Format.IMAGE_PNG, new ClassPathResource("/test.png")))
+			.call()
+			.content();
+
+		logger.info(response);
+		assertThat(response).containsAnyOf("banan", "apple", "basket");
+	}
+
+	@Test
+	void videoMultiModalityTest() throws IOException {
+
+		String response = ChatClient.create(chatModel)
+			.prompt()
+			.user(u -> u.text("Explain what do you see in this video?")
+				.media(Media.Format.VIDEO_MP4, new ClassPathResource("/test.video.mp4")))
+			.call()
+			.content();
+
+		logger.info(response);
+		assertThat(response).containsAnyOf("baby", "small");
+		assertThat(response).containsAnyOf("chicks", "chickens");
+	}
+
+	public static record WeatherRequest(String location, String unit) {
+	}
+
+	public static record WeatherResponse(int temp, String unit) {
+	}
+
+	@Test
+	void functionCallTest() {
+
+		// @formatter:off
+		String response = ChatClient.create(this.chatModel).prompt()
+				.user("What's the weather like in San Francisco, Tokyo, and Paris?  Use Celsius.")
+				.functions(FunctionCallback.builder()
+					.function("getCurrentWeather", (WeatherRequest request) -> {
+						if (request.location().contains("Paris")) {
+							return new WeatherResponse(15, request.unit());
+						}
+						else if (request.location().contains("Tokyo")) {
+							return new WeatherResponse(10, request.unit());
+						}
+						else if (request.location().contains("San Francisco")) {
+							return new WeatherResponse(30, request.unit());
+						}
+			
+						throw new IllegalArgumentException("Unknown location: " + request.location());
+					})
+					.inputType(WeatherRequest.class)
```

Comment:
I added `.description("Get the weather for a city in Celsius")`

note, that the description we normally use for this didn't work `.description("Get the weather in location")`